include/xtensor-io/xnpz.hpp

```diff
@@ -39,6 +39,98 @@ namespace xt
 {
     using namespace std::string_literals;
 
+    namespace detail
+    {
+#pragma pack(push, 2)
+        struct zip_local_header
+        {
+            char sig[4];
+            uint16_t version;
+            uint16_t gp_flags;
+            uint16_t compression_method;
+            uint16_t last_modification_time;
+            uint16_t last_modification_date;
+            uint32_t crc32;
+            uint32_t compressed_size;
+            uint32_t uncompressed_size;
+            uint16_t filename_len;
+            uint16_t extra_field_len;
+
+            auto signature() const { return std::string(sig, sizeof(sig)); }
+        };
+#pragma pack(pop)
+
+        struct extensible_data_field
+        {
+            uint16_t header_id;
+            uint16_t data_size;
+        };
+
+#pragma pack(push, 4)
```

Comment:
Same question here

Reply:
I think we remove the padding so the binary schema is the same as in the file.

Reply:
so that you can reinterpret_cast the first couple of bytes and see if everything falls into place ... that's how I understood it -- also the tests are green so I assume it works on OS X and Linux (and MSVC has docs for it as well).

Reply:
This pragma is ignored on Linux and OSX, so having the tests green probably means that this padding is not really required.

Reply:
Some information about these packed structs: I need to remove the padding between data members and the padding in the end because ZIP isn't designed to take alignment into account.  The `alignas` specifier can only increase alignment, not remove or reduce alignment.

https://godbolt.org/z/CX22QZ (removing the pragma causes `sizeof` to increase)

There is still a problem remaining: The `crc32` member and the two subsequent members are unaligned given their type.  It doesn't work on old ARM chips.  But it may not worth a fix (for example, split the local header into two halves) before we see the first bug report.

Reply:
Thanks @zhihaoy for the explanation. I agree with you regarding the remaining members, let's wait and see if someone complains about them.